README.template.md

```diff
@@ -0,0 +1,237 @@
+
+{{ ossHeader }}
+
+# Momento Go-Redis Compatibility Client
+
+## What and Why?
+
+This project provides a Momento-backed implementation of [go-redis](hhttps://github.com/redis/go-redis).
+The goal is to provide a drop-in replacement for [go-redis](hhttps://github.com/redis/go-redis) so that you can
+use the same code with either a Redis server or with the Momento Cache service!
+
+You can use Momento as your cache engine for any Go project that support a redis-backed cache.
+
+## Usage
+
+To switch your existing `go-redis` application to use Momento, you only need to change the code where you construct your client object:
+
+### With go-redis client
+
+```go
+package redis
+
+import (
+  "github.com/redis/go-redis/v9"
+)
+
+func initRedisClient() redis.Cmdable {
+  // Replace these values with your Redis server's details
+  REDIS_HOST := "my.redis-server.com"
+  REDIS_PORT := "6379"
+  // Create a Redis client
+  redisClient := redis.NewClient(&redis.Options{Addr: REDIS_HOST + ":" + REDIS_PORT})
+  return redisClient
+}
+```
+
+### With Momento's go-redis compatibility client
+
+```go
+package redis
+
+import (
+  "context"
+  "github.com/momentohq/client-sdk-go/auth"
+  "github.com/momentohq/client-sdk-go/config"
+  "github.com/momentohq/client-sdk-go/momento"
+  momentoredis "github.com/momentohq/momento-go-redis-client/momento-redis"
+  "github.com/redis/go-redis/v9"
+  "time"
+)
+
+func initRedisClient() redis.Cmdable {
+  credential, eErr := auth.NewEnvMomentoTokenProvider("MOMENTO_API_KEY")
+  if eErr != nil {
+    panic("Failed to initialize credentials through API key " + eErr.Error())
+  }
+  cacheClient, cErr := momento.NewCacheClient(config.LaptopLatest(), credential, 60*time.Second)
+  if cErr != nil {
+    panic("Failed to initialize Momento cache client " + cErr.Error())
+  }
+  // create cache; it resumes execution normally incase the cache already exists
```

Comment:
Let's remove this createCache stuff. We don't really want people to be calling this every time they instantiate a client. Feel free to add a sentence somewhere outside the code that indicates that they need to have created the cache (via console or whatever).